readthedocs/oauth/services/base.py

```diff
@@ -133,6 +133,7 @@ def token_updater(self, token):
 
         def _updater(data):
             token.token = data['access_token']
+            token.token_secret = data.get("refresh_token", "")
```

Comment:
This can probably be just, I think all providers have this field present (even if set to empty)
```suggestion
            token.token_secret = data["refresh_token"]
```

Reply:
```suggestion
            token.token_secret = data.get("refresh_token", "")
            if not token.token_secret:
                log.warning('OAuth token secret missing')
```

Perhaps this makes more sense, to be defensive?

Reply:
GitHab doesn't have a refresh token (but I'm not really sure github does ever hit this logic, since their tokens never expire)

Reply:
Just checked, bitbucket and github are the only ones that hit this logic, and both return a refresh token.